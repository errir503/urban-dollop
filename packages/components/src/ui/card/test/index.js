/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import {
	Card,
	CardBody,
	CardFooter,
	CardHeader,
	CardInnerBody,
} from '../index';

describe( 'props', () => {
	let base;

	beforeEach( () => {
		base = render(
			<Card>
				<CardHeader title="WordPress.org" />
				<CardBody>Code is Poetry.</CardBody>
				<CardFooter>
					<button>Agree</button>
				</CardFooter>
			</Card>
		);
	} );

	test( 'should render correctly', () => {
		expect( base.container.firstChild ).toMatchSnapshot();
	} );

	test( 'should render elevation', () => {
		const { container } = render( <Card elevation /> );
		const { container: withoutElevation } = render(
			<Card elevation={ false } />
		);
		expect( container.firstChild ).toMatchDiffSnapshot(
			withoutElevation.firstChild
		);
	} );

	test( 'should render CardInnerBody', () => {
		const { container } = render(
			<Card>
				<CardHeader title="WordPress.org" />
				<CardInnerBody>Code is Poetry.</CardInnerBody>
				<CardFooter>
					<button>Agree</button>
				</CardFooter>
			</Card>
		);
		expect( container.firstChild ).toMatchDiffSnapshot(
			base.container.firstChild
		);
	} );
} );
