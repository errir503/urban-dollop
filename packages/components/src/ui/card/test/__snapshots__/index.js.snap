// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props should render CardInnerBody 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -11,13 +11,13 @@
        data-wp-c16t="true"
        data-wp-component="CardHeader"
        title="WordPress.org"
      />
      <div
-       class="css-yc36db-InnerBody components-card-inner-body css-1mm2cvy-View em57xhy0"
+       class="components-scrollable components-card-body css-1kkgl6g-Scrollable-scrollableScrollbar-scrollY-Body-borderRadius css-1mm2cvy-View em57xhy0"
        data-wp-c16t="true"
-       data-wp-component="CardInnerBody"
+       data-wp-component="CardBody"
      >
        Code is Poetry.
      </div>
      <div
        class="components-flex components-card-footer css-g6s6xo-Flex-ItemsRow-borderRadius-headerFooter-medium css-1mm2cvy-View em57xhy0"
`;

exports[`props should render correctly 1`] = `
.emotion-9 {
  height: 100%;
}

.emotion-18 {
  background-color: #fff;
  color: #000;
  position: relative;
  box-shadow: 0 0 0 1px rgba(0,0,0,0.1);
  outline: none;
  border-radius: 2px;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  border-bottom: 1px solid;
  border-color: rgba(0,0,0,0.1);
  min-height: 44px;
  padding-bottom: calc(4px * 1);
  padding-left: calc(4px * 3);
  padding-right: calc(4px * 3);
  padding-top: calc(4px * 1);
}

.emotion-0 > * + *:not(marquee) {
  margin-left: calc(4px * 2);
}

.emotion-0 > * {
  min-width: 0;
}

.emotion-0:first-of-type {
  border-top-left-radius: 2px;
  border-top-right-radius: 2px;
}

.emotion-0:last-of-type {
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
}

.emotion-3 {
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  height: auto;
  max-height: 100%;
  padding: calc(4px * 3) calc(4px * 3);
}

@media only screen and ( min-device-width:40em ) {
  .emotion-3::-webkit-scrollbar {
    height: 12px;
    width: 12px;
  }

  .emotion-3::-webkit-scrollbar-track {
    background-color: transparent;
  }

  .emotion-3::-webkit-scrollbar-track {
    background: rgba(0,0,0,0.04);
    border-radius: 8px;
  }

  .emotion-3::-webkit-scrollbar-thumb {
    background-clip: padding-box;
    background-color: rgba(0,0,0,0.2);
    border: 2px solid rgba( 0,0,0,0 );
    border-radius: 7px;
  }

  .emotion-3:hover::-webkit-scrollbar-thumb {
    background-color: rgba(0,0,0,0.5);
  }
}

.emotion-3:first-of-type {
  border-top-left-radius: 2px;
  border-top-right-radius: 2px;
}

.emotion-3:last-of-type {
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
  border-color: rgba(0,0,0,0.1);
  min-height: 44px;
  padding-bottom: calc(4px * 1);
  padding-left: calc(4px * 3);
  padding-right: calc(4px * 3);
  padding-top: calc(4px * 1);
}

.emotion-6 > * + *:not(marquee) {
  margin-left: calc(4px * 2);
}

.emotion-6 > * {
  min-width: 0;
}

.emotion-6:first-of-type {
  border-top-left-radius: 2px;
  border-top-right-radius: 2px;
}

.emotion-6:last-of-type {
  border-bottom-left-radius: 2px;
  border-bottom-right-radius: 2px;
}

.emotion-12 {
  background: transparent;
  display: block;
  margin: 0 !important;
  pointer-events: none;
  position: absolute;
  will-change: box-shadow;
  border-radius: inherit;
  bottom: 0;
  box-shadow: 0 1px 2px 0 rgba(0 ,0,0,0.05);
  opacity: 1;
  left: 0;
  right: 0;
  top: 0;
  -webkit-transition: box-shadow 200ms cubic-bezier(0.08,0.52,0.52,1);
  transition: box-shadow 200ms cubic-bezier(0.08,0.52,0.52,1);
  border-radius: 2px;
}

.emotion-15 {
  background: transparent;
  display: block;
  margin: 0 !important;
  pointer-events: none;
  position: absolute;
  will-change: box-shadow;
  border-radius: inherit;
  bottom: 0;
  box-shadow: 0 2px 4px 0 rgba(0 ,0,0,0.1);
  opacity: 1;
  left: 0;
  right: 0;
  top: 0;
  -webkit-transition: box-shadow 200ms cubic-bezier(0.08,0.52,0.52,1);
  transition: box-shadow 200ms cubic-bezier(0.08,0.52,0.52,1);
  border-radius: 2px;
}

<div
  class="components-surface components-card emotion-18 emotion-1 emotion-2"
  data-wp-c16t="true"
  data-wp-component="Card"
>
  <div
    class="emotion-9 emotion-1 emotion-2"
  >
    <div
      class="components-flex components-card-header emotion-0 emotion-1 emotion-2"
      data-wp-c16t="true"
      data-wp-component="CardHeader"
      title="WordPress.org"
    />
    <div
      class="components-scrollable components-card-body emotion-3 emotion-1 emotion-2"
      data-wp-c16t="true"
      data-wp-component="CardBody"
    >
      Code is Poetry.
    </div>
    <div
      class="components-flex components-card-footer emotion-6 emotion-1 emotion-2"
      data-wp-c16t="true"
      data-wp-component="CardFooter"
    >
      <button>
        Agree
      </button>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="components-elevation emotion-12 emotion-1 emotion-2"
    data-wp-c16t="true"
    data-wp-component="Elevation"
  />
  <div
    aria-hidden="true"
    class="components-elevation emotion-15 emotion-1 emotion-2"
    data-wp-c16t="true"
    data-wp-component="Elevation"
  />
</div>
`;

exports[`props should render elevation 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -6,16 +6,16 @@
    <div
      class="css-l50fc7-Content css-1mm2cvy-View em57xhy0"
    />
    <div
      aria-hidden="true"
-     class="components-elevation css-109wvzr-Elevation css-1mm2cvy-View em57xhy0"
+     class="components-elevation css-xtff41-Elevation css-1mm2cvy-View em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Elevation"
    />
    <div
      aria-hidden="true"
-     class="components-elevation css-1ajhhcx-Elevation css-1mm2cvy-View em57xhy0"
+     class="components-elevation css-hdn3qt-Elevation css-1mm2cvy-View em57xhy0"
      data-wp-c16t="true"
      data-wp-component="Elevation"
    />
  </div>
`;
